client/app/scripts/superdesk-authoring/authoring.js

```diff
@@ -894,7 +894,12 @@
                 $scope.close = function() {
                     _closing = true;
                     authoring.close($scope.item, $scope.origItem, $scope.dirty).then(function() {
-                        $location.url($scope.referrerUrl);
+                        if (sessionStorage.getItem('previewUrl')) {
+                            $location.url(sessionStorage.getItem('previewUrl'));
+                            sessionStorage.removeItem('previewUrl');
```

Comment:
I was referring to the `sessionStorage.removeItem('previewUrl')` line - does it get executed when `$location.url(...)` on the previous line already does a redirect?